File: Phase2/src/main/java/controller/SearchEngine.java
package controller;

import java.util.Set;

import model.DocumentFile;

public class SearchEngine {
    private final QueryHandler queryHandler;

    public SearchEngine(QueryHandler queryHandler) {
        this.queryHandler = queryHandler;
    }

    public String search(String searchQuery) {
        Set<DocumentFile> resultList = queryHandler.search(searchQuery);
        return docListToString(resultList);
    }

    public static String docListToString(Set<DocumentFile> docList) {
        var result = new StringBuilder();
        for (DocumentFile file : docList)
            result.append(file.toString());
        return result.toString();
    }
}

File: Phase8/SearchEngine/SearchEngine/Controller/SearchEngine.cs
using System.Collections.Generic;
using SearchEngine.Controller.DataBase;
using SearchEngine.Model;
using SearchEngine.Model.Entities;

namespace SearchEngine.Controller
{
    public class SearchEngine : ISearchEngine
    {
        private IDbHandler _index;
        private List<IDocument> _documentsList;
        private Context _context;

        public SearchEngine()
        {
            _index = new DbHandler();
            _documentsList = new List<IDocument>();
            _context = new Context();
        }

        public void AddDirPath(string dirPath)
        {
            throw new System.NotImplementedException();
        }

        public List<IDocument> Search(string searchQuery)
        {
            var queryHandler = new QueryHandler(searchQuery);
            return queryHandler.OperateOnQuery(_index);
        }
    }
}